hbase/src/main/java/com/yahoo/ycsb/db/HBaseClient10.java

```diff
@@ -246,17 +266,22 @@ public int read(String table, String key, Set<String> fields, HashMap<String,Byt
             //do nothing for now...need to understand HBase concurrency model better
             return ServerError;
         }
-
-        if (r.isEmpty()) {
+        if (multiGetBatchSize == 0) {
+          if (r[0].isEmpty()) {
             return NoMatchingRecord;
+          }
         }
-        for (Cell c : r.listCells()) {
-            result.put(Bytes.toString(CellUtil.cloneQualifier(c)),
-                    new ByteArrayByteIterator(CellUtil.cloneValue(c)));
-            if (_debug) {
-                System.out.println("Result for field: "+Bytes.toString(CellUtil.cloneQualifier(c))+
-                        " is: "+Bytes.toString(CellUtil.cloneValue(c)));
+        for (Result res : r) {
+          if(!res.isEmpty()) {
+            for (Cell c : res.listCells()) {
+              result.put(Bytes.toString(CellUtil.cloneQualifier(c)),
```

Comment:
For the verification part to be successful we need to make the result HashMap to a Vector of hashmap. But doing that will have an impact on the signature of the read() API in the base class and it will have an impact on the other DB impl also.  So can we add an abstract class for HBase and make HBaseclient and HBaseClient10 to implement that new abstract class?